src/main/resources/mustache-templates/phenotypes.html

```diff
@@ -0,0 +1,99 @@
+{{>layout/header}}
+
+<div class="container vertical-space">
+	<h3>Patient Info</h3>
+	<ul class="list-unstyled">
+		{{#patient}}
+		<li>Name: <span>{{firstName}} {{lastName}}</span></li>
+		<li>Id: <span id="patient_id">{{id}}</span></li> 
+		{{/patient}}
+	</ul>
+	<h3>Phenotypes</h3>
+	<table id="hposummary" class="table table-bordered" style="width: 100%">
+		<thead class="thead-light">
+			<tr>
+				<th></th>
+				<th>HPO Term Name</th>
+				<th>HPO Term Id</th>
+				<th># Observations</th>
+				<th>First Date</th>
+				<th>Last Date</th>
+			</tr>
+		</thead>
+	</table>
+</div>
+
+
+<script type="text/javascript"
+	src="{{req.contextPath}}/webjars/jquery/jquery.min.js"></script>
+<script type="text/javascript"
+	src="{{req.contextPath}}/webjars/datatables/js/jquery.dataTables.min.js"></script>
+<script type="text/javascript"
+	src="{{req.contextPath}}/webjars/datatables/js/dataTables.bootstrap4.min.js"></script>
+<script type="text/javascript">
+	function showLabs(d) {
+		// `d` is the original data object for the row
+		let table = "<div class='card float-left'>"
+				+ "<table class='table-bordered sub-table'>"
+				+ "<thead>"
+				+ "<tr class='table-warning'><th>FHIR Id</th><th>LOINC</th><th>Description</th><th>Value</th><th>Date</th></tr>"
+				+ "</thead>" + "<tbody>";
+		for (i = 0; i < d.observations.length; i++) {
+			let observation = d.observations[i];
+			table += "<tr>" + "<td>" + observation.fhirId + "</td>" + "<td>"
+					+ observation.loincId + "</td>" + "<td>"
+					+ observation.description + "</td>" + "<td>"
+					+ observation.value + "</td>" + "<td>" + observation.date
+					+ "</td>" + "</tr>";
+		}
+		table += "</tbody></table></div>";
+		return table;
+	}
+
+	let patientId = $("#patient_id").html();
+
+	$(document).ready(function() {
+		// Set up table sorting
+		let table = $("#hposummary").DataTable({
+			ajax : {
+				"url" : "/summary/" + patientId
+			},
+			info: true,
+			paging : true,
+			searching : true,
+			columns : [ {
+				"className" : "details-control",
+				"orderable" : false,
+				"data" : null,
+				"defaultContent" : ''
+				}, 
+				{ "data" : "hpoTermName" }, 
+				{ "data" : "hpoTermId" },
+				{ "data" : "count"},
+				{ "data" : "first" },
+				{ "data" : "last"}
+			],
+			aaSorting : []
+		// Do not sort any columns initially
+		});
+
+		// Add event listener for opening and closing details
+		$('#hposummary tbody').on('click', 'td.details-control', function() {
+			let tr = $(this).closest('tr');
+			let row = table.row(tr);
+
+			if (row.child.isShown()) {
+				// This row is already open - close it
+				row.child.hide();
+				tr.removeClass('shown');
+			} else {
+				// Open this row
+				row.child(showLabs(row.data())).show();
+				tr.addClass('shown');
+			}
+		});
+
```

Comment:
Does this content need to be created dynamically, or could you just create it in the mustache view and have it hidden?

Reply:
My original design tried to create all content in the mustache view, but it doesn't play nicely with DataTables and its "child" function. I used this example as a guide, but couldn't get it to work without implementing the request as ajax: https://datatables.net/examples/api/row_details.html

Reply:
I guess in general maybe we should consider whether the tables with child details is a good pattern for this use case. This UI pattern tends to not scale well on different size devices. I'm not sure about how it works with accessibility, but it seems like semantically the notion of nested tables can get confusing. We could brainstorm with Liz some other approaches.